src/app/Topology/Shared/Entity/utils.ts

```diff
@@ -0,0 +1,337 @@
+      throw new Error(`Unsupported resource: ${resourceType}`);
+  }
+};
+
+export const getResourceRemovedEvent = (resourceType: TargetOwnedResourceType | TargetRelatedResourceType) => {
+  switch (resourceType) {
+    case 'activeRecordings':
+      return [NotificationCategory.ActiveRecordingDeleted, NotificationCategory.SnapshotDeleted];
+    case 'archivedRecordings':
+      return [NotificationCategory.ArchivedRecordingDeleted];
+    case 'archivedUploadRecordings':
+      return [NotificationCategory.ArchivedRecordingDeleted];
+    case 'eventTemplates':
+      return [NotificationCategory.TemplateDeleted];
+    case 'eventTypes':
+      return [];
+    case 'agentProbes':
+      return [NotificationCategory.ProbesRemoved];
+    case 'automatedRules':
+      return [NotificationCategory.RuleDeleted];
+    case 'credentials':
+      return [NotificationCategory.CredentialsDeleted, NotificationCategory.TargetCredentialsDeleted];
+    default:
+      throw new Error(`Unsupported resource: ${resourceType}`);
+  }
+};
+
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+export type PatchFn = (arr: ResourceTypes[], eventData: any, removed?: boolean) => Observable<ResourceTypes[]>;
+
+export const getResourceListPatchFn = (
+  resourceType: TargetOwnedResourceType | TargetRelatedResourceType,
+  { target }: TargetNode,
+  apiService: ApiService
+): PatchFn => {
+  switch (resourceType) {
+    case 'activeRecordings':
+    case 'archivedRecordings':
+    case 'archivedUploadRecordings':
+      return (arr: Recording[], eventData: any, removed?: boolean) => {
+        const recording: Recording = eventData.message.recording;
+        let newArr = arr.filter((r) => r.name !== recording.name);
+        if (!removed) {
+          newArr = newArr.concat([recording]);
+        }
+        return of(newArr);
+      };
+    case 'eventTemplates':
+      return (arr: EventTemplate[], eventData: any, removed?: boolean) => {
+        const template: EventTemplate = eventData.message.template;
+        let newArr = arr.filter((r) => r.name !== template.name);
+        if (!removed) {
+          newArr = newArr.concat([template]);
+        }
+        return of(newArr);
+      };
+    case 'agentProbes':
+      return (arr: EventProbe[], eventData: any, removed?: boolean) => {
+        // Only support remove all
+        if (removed) {
+          return of([]);
+        }
+        const probes = (eventData.message.events as EventProbe[]) || [];
+        const probeIds = probes.map((p) => p.id);
+        return of([...arr.filter((probe) => !probeIds.includes(probe.id)), ...probes]);
+      };
+    case 'automatedRules':
+      return (arr: Rule[], eventData: any, removed?: boolean) => {
+        const rule: Rule = eventData.message;
+
+        return apiService.isTargetMatched(rule.matchExpression, target).pipe(
+          map((ok) => {
+            if (ok) {
+              // return apiService.isTargetMatched(credential.matchExpression, )
```

Comment:
What's this comment for?

Reply:
Another copy and paste but forgot to clean up :D